components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/util/ActionMgtV2AuditLoggerUtil.java

```diff
@@ -0,0 +1,248 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONObject;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.Authentication;
+import org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.utils.AuditLog;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACCESS_TOKEN_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACTION_DESCRIPTION_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACTION_ID_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACTION_NAME_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACTION_STATUS_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACTION_TYPE_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ACTIVATE_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ADD_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.API_KEY_HEADER_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.API_KEY_VALUE_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.DEACTIVATE_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.DELETE_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ENDPOINT_AUTHENTICATION_SCHEME_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ENDPOINT_CONFIG_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.ENDPOINT_URI_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.PASSWORD_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.UPDATE_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.USERNAME_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.jsonObjectToMap;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.triggerAuditLogEvent;
+
+/**
+ * Action management V2 audit logger util class.
+ */
+public class ActionMgtV2AuditLoggerUtil {
```

Comment:
ActionManagementAuditLogger

Reply:
IMO, Using v1 of cetral logger or v2 is an implementation decision of this class. So let's keep it that way

Reply:
Refactored with a105238b1c38f7405ecf43ecaa9c07b9b0de7b7f